application/src/main/java/com/sanction/lightning/facebook/FacebookService.java

```diff
@@ -102,17 +104,30 @@ public FacebookUser getFacebookUser() {
    *
    * @return A boolean to denote that the function worked.
    */
-  public FacebookPhoto publishToFacebook(byte[] inputBytes, String fileName, String message) {
-    FacebookPhoto response;
+  public String publishToFacebook(byte[] inputBytes, String type, String fileName,
+                                  String message, String videoTitle) {
+    JsonObject response;
+    List<Parameter> parameters = Lists.newArrayList();
+
+    if (type.equals("photo")) {
+      type = "photos";
+      parameters.add(Parameter.with("message", message));
+    } else if (type.equals("video")) {
+      type = "videos";
+      parameters.add(Parameter.with("description", message));
+      parameters.add(Parameter.with("title", videoTitle));
+    } else {
+      return null;
```

Comment:
You shouldn't need this `else` and `return null` anymore - you check before calling this function.

As a side note, I think we should make an `enum PublishType` or something so we can avoid all those checks, and make the code more readable. We can do that after this PR though.